src/screens/sync/Devices.tsx

```diff
@@ -212,10 +57,39 @@ export const Devices = ({ children }: { children: React.ReactNode }) => {
               <AnimatedEllipsis />
             </Text>
           </View>
-        ) : (
-          children
-        )}
+        ) : mode === "list" ? (
+          <DevicesList
+            peers={peers}
+            onInfoPress={() => {
+              setModalMode({ type: "info" });
+              openSheet();
+            }}
+            onDevicePress={(peer: Peer) => {
+              setModalMode({ type: "device", data: peer });
+              openSheet();
+            }}
+          />
+        ) : null}
       </View>
-    </React.Fragment>
+      <BottomSheetModal
+        ref={sheetRef}
+        disableBackdropPress={modalMode?.type === "device"}
+      >
+        {!modalMode ? null : modalMode.type === "info" ? (
+          <LocalDevicesInfo />
+        ) : modalMode.type === "device" ? (
+          <DeviceBottomSheetContent
+            onClose={() => closeSheet()}
+            onRemoveDevice={() => {
+              setPeers((prev) =>
+                prev.filter(({ id }) => modalMode.data.id !== id)
+              );
+            }}
+            role={role}
+            peer={modalMode.data}
```

Comment:
would it not be simpler to make `peer={peer}`

Right now you are setting '`peer` in `devices`, passing it to `deviceList` and then having `devicelist` pass it to the `modal` which is in `devices`. Since the modal is already in the parent, I don't think there is a need for the modal to access `peer` from the child

Reply:
`Devices` is handling `peers` (not `peer`), which represents all of the peers/devices that exist in the list. The `peer` prop here is taking the specific peer that was pressed in the device list to show information specific to the selected device in the bottom sheet contents

Reply:
OHHH That makes alot more sense

Reply:
sorry, slightly misread your comment. `Devices` needs to know which row in the `DevicesList` was pressed, so that it can use that information to determine what to show in the bottom sheet content